image/src/app/module/main.py

```diff
@@ -34,18 +36,28 @@ def module_main(parsed_data):
             return_body = []
             for data in parsed_data:
                 return_body.append(processData(data))
-
+        # set header if necessary
+        headers = {}
+        if APPLICATION['CONTENT_TYPE_JSON'] == 'yes':
+            headers.update({'Content-Type': 'application/json'})
+        if APPLICATION['AUTHENTICATION_REQUIRED'] == 'yes' and APPLICATION['ACCESS_TOKEN'] != '':
+            headers.update({"Authorization": f"{APPLICATION['ACCESS_TOKEN']}"})
+        if APPLICATION['AUTHENTICATION_API_KEY'] != '':
+            headers.update({'x-api-key': APPLICATION['AUTHENTICATION_API_KEY']})
         if METHOD == "POST":
-            post(url=f"{EGRESS_WEBHOOK_URL}", json=return_body,
-                 headers={'Content-Type': 'application/json'})
-
+            r = post(url=f"{EGRESS_WEBHOOK_URL}", json=return_body, headers=headers)
         elif METHOD == "GET":
             if type(parsed_data) == dict:
-                get(url=f"{EGRESS_WEBHOOK_URL}", params=return_body)
+                r = get(url=f"{EGRESS_WEBHOOK_URL}", params=return_body, headers=headers)
             else:
-                get(url=f"{EGRESS_WEBHOOK_URL}",
-                    params={"data": dumps(return_body)})
-
+                r = get(
+                    url=f"{EGRESS_WEBHOOK_URL}",
+                    params={"data": dumps(return_body)},
+                    headers=headers,
+                )
+        if not r.ok and APPLICATION['ERROR_URL'] != '':
+            info = {"url": EGRESS_WEBHOOK_URL, "data": return_body, "errorCode": r.status_code}
+            post(url=f"{APPLICATION['ERROR_URL']}", json=info)
```

Comment:
What if I don't provide ERROR_URL because I don't have it set up, but I still want to know that my data were not sent to the endpoint? Then we should log the error of transferring the data. You can add
`elif not r.ok: return None, f'Unable to transfer data: {r.status_code}`

This will just log the error to the console.

Reply:
ok, I made the change.